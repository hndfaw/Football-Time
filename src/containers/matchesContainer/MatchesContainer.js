import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import Match from '../../components/match/Match';

export class MatchesContainer extends Component {
  
  render() {
    const {id, todaysMatches} = this.props
    const filterMatchesPerLeague = todaysMatches.filter(match => 
       match.league_id === id
    )

    const match = filterMatchesPerLeague.map(match => {
      return (
        <Match key={match.fixture_id} match={match}/>
      )
    })
    return (
      <div>
        {match}
      </div>
    )
  }
}

MatchesContainer.propTypes = {
  todaysMatches: PropTypes.array
}

export const mapStateToProps = state => ({
  todaysMatches: state.todaysMatches,
})

export default connect(mapStateToProps, null)(MatchesContainer)
