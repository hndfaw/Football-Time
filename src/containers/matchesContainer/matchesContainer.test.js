import React from 'react';
import { MatchesContainer, mapStateToProps } from './MatchesContainer';
import { shallow } from 'enzyme';


describe('MatchesContainer', () => {

  it('should match the snapshot', () => {
    let mockMatch = [{
      awayTeamLogo: "https://www.api-football.com/public/teams/3700.png",
      awayTeamName: "Always Ready",
      event_date: "2019-07-31",
      fixture_id: 201290,
      homeTeamLogo: "https://www.api-football.com/public/teams/3709.png",
      homeTeamName: "Royal Pari",
      league_id: 476,
      statusShort: "PST",
    }] 
    let wrapper = shallow(<MatchesContainer todaysMatches={mockMatch}/>)
    expect(wrapper).toMatchSnapshot()
  })

})