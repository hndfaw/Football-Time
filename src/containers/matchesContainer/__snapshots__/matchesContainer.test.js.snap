// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MatchesContainer should match the snapshot 1`] = `<div />`;
